### PR TITLE
[libsrtp] Added openssl feature flag for enabling OpenSSL support

### PR DESCRIPTION
**Pull request by asantoni**
This pull request adds a feature flag for the existing libsrtp port, to allow it to be built with OpenSSL support. OpenSSL support is necessary in order for libsrtp to support the AES GCM modes, which you won't realize until you've debugged it. 🙃 

The changes are minimal and hopefully correct, but please let me know if any modifications are needed!

To use the feature flag, you can add "openssl" to the list of features on the dependency (as per the vcpkg docs, same way as any other feature):

```
Your vcpkg.json:
....
  "dependencies": [
    ...
    {"name": "libsrtp", "features": ["openssl"]},
    "openssl",
...
```

### CODE REVIEW

**Comment by asantoni** [comment]
@microsoft-github-policy-service agree company="Radio Mast, Inc."

File: ports/libsrtp/portfile.cmake [hunk 1]
```diff
@@ -7,10 +7,16 @@ vcpkg_from_github(
...
+    set(ENABLE_OPENSSL ON)
+endif()
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
+        -DENABLE_OPENSSL=${ENABLE_OPENSSL}
         -DTEST_APPS=OFF
 )
```
Comment (Cheney-W) [inline ports/libsrtp/portfile.cmake:21]:
```suggestion
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        openssl ENABLE_OPENSSL
)

vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
        -DTEST_APPS=OFF
        ${FEATURE_OPTIONS}
)

```

Comment (Cheney-W) [reply ports/libsrtp/portfile.cmake:21]: https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_check_features

Comment (Cheney-W) [reply ports/libsrtp/portfile.cmake:21]: Moving the PR to draft. Please set it to "ready for review" once changes have been applied. Thank you!

**Comment by asantoni** [comment]
@Cheney-W Thanks for the review and suggestions. I have implemented and tested your changes, and so I think this PR is ready for another review.

**Comment by Cheney-W** [comment]
Please modify the vcpkg.json as below:
```
{
  "name": "libsrtp",
  "version": "2.5.0",
  "port-version": 1,
  "description": "This package provides an implementation of the Secure Real-time Transport Protocol (SRTP), the Universal Security Transform (UST), and a supporting cryptographic kernel.",
...
```

File: ports/libsrtp/portfile.cmake [hunk 2]
```diff
@@ -7,13 +7,20 @@ vcpkg_from_github(
...
+
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
         -DTEST_APPS=OFF
+        ${FEATURE_OPTIONS}
 )
 
+
```
Comment (Cheney-W) [inline ports/libsrtp/portfile.cmake:23]:
```suggestion
```
Please delete extra blank lines.

File: ports/libsrtp/vcpkg.json
Comment (kevle) [inline ports/libsrtp/vcpkg.json:1]: Port version should probably be bumped
`"port-version": 1`

Comment (Cheney-W) [reply ports/libsrtp/vcpkg.json:1]: Yes, I have mentioned it in this [comment](https://github.com/microsoft/vcpkg/pull/36135#issuecomment-1892994714).

**Comment by asantoni** [comment]
Thanks for the continued reviews @Cheney-W @kevle. I have made those two tweaks and updated the version database. This should be ready for review again.

File: versions/l-/libsrtp.json [hunk 3]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "27c325c9851b5c07e172e74e312e8070af7b9d0b",
+      "git-tree": "15577c99d2d078770d2e07d2bc8cce770797adb8",
+      "version": "2.5.0",
+      "port-version": 1
+    },
+    {
+      "git-tree": "0f0b21bc9987edfc4a76c49ecdcaa04d67c7d34c",
```
Comment (Cheney-W) [inline versions/l-/libsrtp.json:9]:
```suggestion
      "git-tree": "27c325c9851b5c07e172e74e312e8070af7b9d0b",
```

**Review by Cheney-W — CHANGES_REQUESTED** [review]

**Comment by Cheney-W** [comment]
If you first modify the port file and commit them, you should run `vcpkg x-add-version libsrtp`;
if you modify the port file and commit them again, you should run vcpkg `x-add-version libsrtp --overwrite-version`

**Comment by Cheney-W** [comment]
Please click the `Ready for review` button when you think this PR is done.

**Review by data-queue — APPROVED** [review]

**Comment by asantoni** [comment]
Aha, thanks @Cheney-W. I forgot to add `--overwrite-version` last time. Should be ready for review now.

**Review by Cheney-W — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Pull request merged**